jsapp/js/components/processing/analysis/list/analysisQuestionRow.component.tsx

```diff
@@ -10,53 +10,181 @@ import CommonHeader from '../responseForms/commonHeader.component';
-          return <CommonHeader uid={question.uid} />;
-        }
-        case 'qual_select_multiple': {
-          return <SelectMultipleResponseForm uid={question.uid} />;
-        }
-        case 'qual_select_one': {
-          return <SelectOneResponseForm uid={question.uid} />;
-        }
-        case 'qual_tags': {
-          return <TagsResponseForm uid={question.uid} />;
-        }
-        default: {
-          return <DefaultResponseForm uid={question.uid} />;
+  // Get the question data from state (with safety check)
+  const question = findQuestion(props.uid, analysisQuestions?.state);
+  if (!question) {
+    return null;
+  }
+
+  const isDragDisabled = analysisQuestions?.state.isPending;
+
+  const previewRef = useRef<HTMLLIElement>(null);
+  const dragRef = useRef<HTMLDivElement>(null);
+
+  const [{handlerId}, drop] = useDrop<
+    DragItem,
+    unknown,
+    {handlerId: Identifier | null}
+  >({
+    accept: DND_TYPES.ANALYSIS_QUESTION,
+    collect(monitor) {
+      return {
+        handlerId: monitor.getHandlerId(),
+      };
+    },
+    hover(item: DragItem, monitor) {
+      if (!previewRef.current) {
+        return;
+      }
+      const dragIndex = item.index;
+      const hoverIndex = props.index;
+
+      // Don't replace items with themselves
+      if (dragIndex === hoverIndex) {
+        return;
+      }
+
+      // Determine rectangle on screen
+      const hoverBoundingRect = previewRef.current?.getBoundingClientRect();
+
+      // Get vertical middle
+      const hoverMiddleY =
+        (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+      // Determine mouse position
+      const clientOffset = monitor.getClientOffset();
+
+      // Get pixels to the top
+      const hoverClientY = (clientOffset as XYCoord).y - hoverBoundingRect.top;
+
+      // Only perform the move when the mouse has crossed half of the items height
+      // When dragging downwards, only move when the cursor is below 50%
+      // When dragging upwards, only move when the cursor is above 50%
+
+      // Dragging downwards
+      if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+        return;
+      }
+
+      // Dragging upwards
+      if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+        return;
+      }
+
+      // Time to actually perform the action
+      props.moveRow(props.uid, dragIndex, hoverIndex);
+
+      // Note: we're mutating the monitor item here!
+      // Generally it's better to avoid mutations,
+      // but it's good here for the sake of performance
+      // to avoid expensive index searches.
+      item.index = hoverIndex;
+    },
+  });
+
+  const [{isDragging}, drag, preview] = useDrag({
+    type: DND_TYPES.ANALYSIS_QUESTION,
+    item: () => {
+      return {id: props.uid, index: props.index};
+    },
+    canDrag: !isDragDisabled,
+    collect: (monitor) => {
+      return {
+        isDragging: monitor.isDragging(),
+      };
+    },
+    end: () => {
```

Comment:
Dragging an item to an ineligible dropzone (itself, outside the list) sends an update response to the (fake) API. It might be nice to short-circuit that by calling `monitor.didDrop()` inside of `end(item, monitor)`.